Scenarios/AKS-Secure-Baseline-PrivateCluster/Terraform/04-Network-Hub/modules/compute-Win/compute-win.tf

```diff
@@ -0,0 +1,111 @@
+resource "azurerm_windows_virtual_machine" "compute" {
+
+  name                            = var.server_name
+  location                        = var.location
+  resource_group_name             = var.resource_group_name
+  size                            = var.vm_size
+  admin_username                  = var.admin_username
+  admin_password                  = var.admin_password
+  tags                            = var.tags
+
+  network_interface_ids = [
+    azurerm_network_interface.compute.id
+  ]
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = var.storage_account_type
+  }
+
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2019-Datacenter"
+    version   = "latest"
+  }
+
+  boot_diagnostics {
+    storage_account_uri = null
+  }
+}
+
+resource "azurerm_network_interface" "compute" {
+
+  name                          = "${var.server_name}-nic"
+  location                      = var.location
+  resource_group_name           = var.resource_group_name
+  enable_accelerated_networking = var.enable_accelerated_networking
+
+  tags = var.tags
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = var.vnet_subnet_id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+locals {
+  
+  set_location         = "Set-Location -Path C:\\Users\\${var.admin_username}"
+  install_az_cli       = "$ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; rm AzureCLI.msi"
+  start_sleep          = "Start-Sleep -Seconds 120"
+  aks_cli              = "curl.exe -LO https://dl.k8s.io/release/v1.26.0/bin/windows/amd64/kubectl.exe"
```

Comment:
Curious: was `dl.k8s.io` this already allowed in the egress firewall?  Is the jumpbox not internet-egressing through the firewall?

Reply:
I will check the rules, it might have but as we have deployed the Bastion service, Client machines will be accessed through the Bastion service. We can discuss keeping the ingress rule on FW or remove.

Reply:
I wasn't talking about ingress at all, my question was exclusively about egress.

Reply:
Will check on egress as well, if not though fw then i will configure it.

Reply:
Yes, egress is allowed for the JB, the code is executed as a TF extension.
FW app rule code snip config:
application_rule_collection {
    name     = "app_rule_collection2"
    priority = 206
    action   = "Allow"
    rule {
      name = "app_rule_collection1_rule2"
      protocols {
        type = "Http"
        port = 80
      }
      protocols {
        type = "Https"
        port = 443
      }
      source_addresses  = ["10.1.0.0/16"]